game/scripts/World.gd

```diff
@@ -1,13 +1,28 @@
 extends Node2D
 
 @onready var stick: Weapon = $Player/stick
+@onready var inventory_interface: Control = $UI/InventoryInterface
+@onready var player: Player = $Player
 
 # Called when the node enters the scene tree for the first time.
 func _ready():
-	pass # Replace with function body.
+	inventory_interface.set_player_inventory_data(player.inventory_data)
+	player.toggle_inventory.connect(toggle_inventory_interface)
 
 
 # Hack for now before having equipment associated with the player
 func _process(delta):
 	if Input.is_action_just_pressed("ui_accept"):
 		stick.use0()
+
+func toggle_inventory_interface() -> void:
```

Comment:
Also fine for now but it feels weird to me to have this interface be a part of the World node. Is that the typical design pattern? feels like having some inventoryHandler being a chid of the player makes more sense to me? IDK or a set of UI nodes?

Reply:
^ I say this having this shitty stick input code here too lol